### PR TITLE
Allow Pallet to use all Cask metadata

### PR DESCRIPTION
**Pull request by sambrightman**
As discussed on #37 (and also causing #36) this is my attempt to be able to use full Cask dependency information. Tests are included.

### CODE REVIEW

File: pallet.el [hunk 1]
```diff
@@ -187,10 +184,29 @@ use `pallet--package-archives-copy' if USE-COPY is true."
...
+    (-when-let (fetcher (cask-dependency-fetcher package))
+      (nconc depend-args (list fetcher))
+      (let ((url (cask-dependency-url package)))
+        (nconc depend-args (list url)))
+      (-when-let (ref (cask-dependency-ref package))
+        (nconc depend-args (list :ref ref)))
+      (-when-let (branch (cask-dependency-branch package))
+        (nconc depend-args (list :branch branch)))
+      (-when-let (files (cask-dependency-files package))
+        (nconc depend-args (list :files files))))
```
Comment (rdallasgray) [inline pallet.el]: Why use `list` rather than quoting?

**Comment by rdallasgray** [comment]
The tests all pass here -- nice contribution, many thanks. If we can clear up the `list` vs quoting issue I'll get it merged. I'll also add a credit to the README for the feature.

File: pallet.el (hunk 1 above)
Comment (sambrightman) [inline pallet.el]: This is some of my first Lisp, so I'm not overly familiar standard idioms. Do you mean backquote (with commas)? The arguments do want to be evaluated, which is why I chose list (and keywords naturally evaluate to themselves).

Comment (rdallasgray) [inline pallet.el]: OK -- I think quotes and backpacks are probably more idiomatic, but for my money `list` is cleaner.

**Pull request merged**